scipy/stats/tests/test_multivariate.py

```diff
@@ -1635,6 +1638,187 @@ def test_haar(self):
+            0.0020732579600816823488240725481546,
+            0.00095660371505271429414668515889275,
+            0.00021831953784896498569831346792114,
+            0.00037725616140301147447000396084604
+        ]
+
+    ), (
+
+        # x
+        [
+            [0.9718, 0.1298, 0.8134],
+            [0.4922, 0.5522, 0.7185],
+            [0.3010, 0.1491, 0.5008],
+            [0.5971, 0.2585, 0.8940],
+            [0.5434, 0.5287, 0.9507],
+        ],
+        # mu
+        [-1, 1, 50],
+        # shape
+        [
+            [1.0000, 0.5000, 0.2500],
+            [0.5000, 1.0000, -0.1000],
+            [0.2500, -0.1000, 1.0000],
+        ],
+        # df
+        8,
+        # ans
+        [
+            0.00000000000000069609279697467772867405511133763,
+            0.00000000000000073700739052207366474839369535934,
+            0.00000000000000069522909962669171512174435447027,
+            0.00000000000000074212293557998314091880208889767,
+            0.00000000000000077039675154022118593323030449058,
+        ]
+    )]
+
+    @pytest.mark.parametrize("x, mu, df, shape, ans", MULTIVARIATE_T_TESTS)
+    def test_pdf_correctness(self, x, mu, df, shape, ans):
+        dist = multivariate_t(mu, df, shape, seed=0)
+        val = dist.pdf(x)
+        assert_array_almost_equal(val, ans)
+
+    @pytest.mark.parametrize("x, mu, df, shape, ans", MULTIVARIATE_T_TESTS)
+    def test_logpdf_correct(self, x, mu, df, shape, ans):
+        dist = multivariate_t(mu, df, shape, seed=0)
+        # We already test this elsewhere, but let's explicitly confirm the
+        # PDF is correct since this test requires checking the log of that
+        # value.
+        val1 = dist.pdf(x)
+        assert_array_almost_equal(val1, ans)
+        val2 = dist.logpdf(x)
+        assert_array_almost_equal(np.log(val1), val2)
+
+    # https://github.com/scipy/scipy/issues/10042#issuecomment-576795195
+    def test_mvt_with_df_one_is_cauchy(self):
+        x = [9, 7, 4, 1, -3, 9, 0, -3, -1, 3]
+        val = multivariate_t.pdf(x, df=1)
+        ans = cauchy.pdf(x)
+        assert_array_almost_equal(val, ans)
+
+    def test_mvt_with_high_df_is_approx_normal(self):
+        # `normaltest` returns the chi-squared statistic and the associated
+        # p-value. The null hypothesis is that `x` came from a normal
+        # distribution, so a low p-value represents rejecting the null, i.e.
+        # that it is unlikely that `x` came a normal distribution.
+        P_VAL_MIN = 0.1
+
+        dist = multivariate_t(0, 1, df=100000, seed=1)
+        samples = dist.rvs(size=100000)
+        _, p = normaltest(samples)
+        assert (p > P_VAL_MIN)
+
+        dist = multivariate_t([-2, 3], [[10, -1], [-1, 10]], df=100000,
+                              seed=42)
+        samples = dist.rvs(size=100000)
+        _, p = normaltest(samples)
+        assert ((p > P_VAL_MIN).all())
+
+        dist = multivariate_t(0, 1, df=np.inf, seed=7)
```

Comment:
for `df=np.inf` the returned results should be ***exactly*** normal.

Reply:
Seems related to [this comment](https://github.com/scipy/scipy/pull/12582#discussion_r489905679).

Does this mean adding an explicit check for `df == np.inf` and returning a `multivariate_normal` instance instead, e.g.:

```python
def __call__(self, loc=None, shape=1, df=1, allow_singular=False,
             seed=None):
    """
    Create a frozen multivariate t-distribution. See
    `multivariate_t_frozen` for parameters.

    """
    if df == np.inf:
        return multivariate_normal_frozen(loc=loc, cov=shape, 
                                          allow_singular=allow_singular, seed=seed)
    return multivariate_t_frozen(loc=loc, shape=shape, df=df,
                                 allow_singular=allow_singular, seed=seed)
```

And then adding docs / tests for this?

Reply:
> Seems related to [this comment](https://github.com/scipy/scipy/pull/12582#discussion_r489905679).
> 
> Does this mean adding an explicit check for `df == np.inf` and returning a `multivariate_normal` instance instead, e.g.:
> 
> ```python
> def __call__(self, loc=None, shape=1, df=1, allow_singular=False,
>              seed=None):
>     """
>     Create a frozen multivariate t-distribution. See
>     `multivariate_t_frozen` for parameters.
> 
>     """
>     if df == np.inf:
>         return multivariate_normal_frozen(loc=loc, cov=shape, 
>                                           allow_singular=allow_singular, seed=seed)
>     return multivariate_t_frozen(loc=loc, shape=shape, df=df,
>                                  allow_singular=allow_singular, seed=seed)
> ```

Yes, both here in `__call__` method and maybe in `_rvs` too. Now that I've looked over the PR again it looks like he `pdf` and `cdf`, `logcdf` methods might need to have branches as well. 

> And then adding docs / tests for this?

I think you already have a test here but for other methods where you add a branch you'd want to add a test case.

Reply:
Both the `pdf` and `logpdf` methods delegate to `multivariate_t._logpdf`, which allowed me to place a second check for `df == np.inf` there. This required me to add a `rank` argument to `_logpdf`, which was already being computed by the `_PSD` helper class.

Additionally, I've created a new unit test checking for exact normality when `df == np.inf`. Since `multivariate_normal` is well-tested, I just handle two cases:

1. When `__call__` is called, I confirm the returned object is `multivariate_normal_frozen`.
2. When `pdf` or `logpdf` are called, I verify `multivariate_normal._logpdf` was called via mocking.

This PR doesn't support `cdf` or `logcdf`.

Reply:
> I don't think I need to check for `df == np.inf` for the `rvs` method since the code is effectively:
> 
> ```python
> # when df == np.inf:
> x = np.ones(size)
> z = rng.multivariate_normal(np.zeros(dim), shape, size=size)
> samples = loc + z / np.sqrt(x)[:, None]
> ```

Yes I see that now, should be fine as written. 

> This PR doesn't support `cdf` or `logcdf`.

Got it, was confusing with univariate distributions for some reason. No need for `cdf` or `logcdf` tests here.